### DISCUSSION TITLE
Why is it not recommended to underline file names？

### DISCUSSION THREAD

**Discussion by unf1a9** [zeromicro/go-zero#2601] [Q&A]
![image](https://user-images.githubusercontent.com/32759384/201524780-e48a71e5-1353-4019-860f-01239f09d9a8.png)

![image](https://user-images.githubusercontent.com/32759384/201524867-d530c696-8a31-49c7-81fd-219c50723266.png)

It's hard to see what a file name means without underlines. I've read source code from other good frameworks where the filename was underlined, even if it’s directories may not be separated with underlines.

<img width="497" alt="image" src="https://user-images.githubusercontent.com/32759384/201871926-d93ce4f1-7345-4f98-8edc-17c164f8d4e4.png">

**Comment by kslamph** [comment]
it's a convention. so follow the convention.
if the question is why there is such a convention.
that's because _ has been assigned to some special meaning in golang.
to name a few:
suffixed _darwin will be build only when os is darwin
suffixed _arm64 will be build only for architect arm64
files prefixed with _ will be ignored by go tools;
go test will run suffixed _test go files;

so _ is a "reserved symbol in filename".
